### PR TITLE
Proposal: separate yurt-manager clients

### PR DESCRIPTION
**Pull request by luc99hen**
#### What type of PR is this?
/kind documentation

#### What this PR does / why we need it:
Propose a plan for separating yurt-manager clients.
#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

File: docs/proposals/20240517-separate-yurtmanager-clients.md [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+
+	clientStore.lock.Lock()
+	defer clientStore.lock.Unlock()
+
+	if cli, ok := clientStore.clientsByName[controllerName]; ok {
+		return cli
+	}
+
+	// check if controller-specific ServiceAccount exist
+	_, err := getOrCreateServiceAccount(mgr.GetClient(), "kube-system", controllerName)
```
Comment (rambohe-ch) [inline docs/proposals/20240517-separate-yurtmanager-clients.md:121]: don't use the hard code `kube-system` for namespace, please use the parameter `WorkingNamespace`.

**Comment by rambohe-ch** [comment]
@luc99hen Thank you for raising proposal to improve rbac management of yurt-manager component. and this proposal will reduce the security risk of OpenYurt cluster. moreover, it is easy for users to track the source controller that apply request to kube-apiserver.

I am looking forward to your code pull request.

**Comment by rambohe-ch** [comment]
@luc99hen At the same time, I think you can make a share presentation on the community meeting this week.

File: docs/proposals/20240517-separate-yurtmanager-clients.md [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+
+Considering the additional complexity of the first approach, we choose the second one to override the token directly when building the client for each controller/webhook.
+
+### Solution Introduction
+
+The whole solution consists of two steps: 
+
+1. Generate RBAC
+
+First, prepare RBAC artifacts including role/clusterrole, rolebindidng/clusterrolebinding and serviceaccount for controllers and webhooks in yurt-manager. The specific steps are as following:
```
Comment (rambohe-ch) [inline docs/proposals/20240517-separate-yurtmanager-clients.md:93]: the name of clusterrole/clusterrolebinding/sa should begin with prefix `openyurt:controller:`, so it is convenient for user to know that this resources is used by controllers of openyurt.

File: docs/proposals/20240517-separate-yurtmanager-clients.md (hunk 1 above)
Comment (luc99hen) [reply docs/proposals/20240517-separate-yurtmanager-clients.md:121]: done

File: docs/proposals/20240517-separate-yurtmanager-clients.md (hunk 2 above)
Comment (luc99hen) [reply docs/proposals/20240517-separate-yurtmanager-clients.md:93]: OK, I will take care of this in implementation.

**Review by rambohe-ch — APPROVED** [review]

**Pull request merged**